volatility/framework/plugins/windows/hashdump.py

```diff
@@ -0,0 +1,346 @@
+        key.append(((ord(s[0]) & 0x01) << 6) | (ord(s[1]) >> 2))
+        key.append(((ord(s[1]) & 0x03) << 5) | (ord(s[2]) >> 3))
+        key.append(((ord(s[2]) & 0x07) << 4) | (ord(s[3]) >> 4))
+        key.append(((ord(s[3]) & 0x0F) << 3) | (ord(s[4]) >> 5))
+        key.append(((ord(s[4]) & 0x1F) << 2) | (ord(s[5]) >> 6))
+        key.append(((ord(s[5]) & 0x3F) << 1) | (ord(s[6]) >> 7))
+        key.append(ord(s[6]) & 0x7F)
+        for i in range(8):
+            key[i] = (key[i] << 1)
+            key[i] = self.odd_parity[key[i]]
+
+        return "".join(chr(k) for k in key)
+
+    def decrypt_single_hash(self, rid, hbootkey, enc_hash, lmntstr):
+        (des_k1, des_k2) = self.sid_to_key(rid)
+        d1 = DES.new(des_k1.encode('latin1'), DES.MODE_ECB)
+        d2 = DES.new(des_k2.encode('latin1'), DES.MODE_ECB)
+        md5 = MD5.new()
+
+        md5.update(hbootkey[:0x10] + pack("<L", rid) + lmntstr)
+        rc4_key = md5.digest()
+        rc4 = ARC4.new(rc4_key)
+        obfkey = rc4.encrypt(enc_hash)
+
+        hash = d1.decrypt(obfkey[:8]) + d2.decrypt(obfkey[8:])
+        return hash
+
+
+
+    def get_user_name(self, user, samhive):
+        V = None
+        for v in user.get_values():
+            if v.get_name() == 'V':
+                V = samhive.read(v.Data+4, v.DataLength)
+        if not V:
+            return None
+
+        name_offset = unpack("<L", V[0x0c:0x10])[0] + 0xCC
+        name_length = unpack("<L", V[0x10:0x14])[0]
+        if name_length > len(V):
+            return None
+
+        username = V[name_offset:name_offset + name_length].decode('utf-16-le')
+        return username
+
+    #replaces the dump_hashes method in vol2
+    def _generator(self, syshive, samhive):
+        if syshive == None:
+            print("SYSTEM address is None: Did you use the correct profile?")
+            yield self.none_obj
+        if samhive == None:
+            print("SAM address is None: Did you use the correct profile?")
+            yield self.none_obj
+        bootkey = self.get_bootkey(syshive)
+        hbootkey = self.get_hbootkey(samhive, bootkey)
+        if hbootkey:
+            for user in self.get_user_keys(samhive):
+                ret = self.get_user_hashes(user, samhive, hbootkey)
+                if not ret:
+                    yield self.none_obj
+                else:
+                    lmhash, nthash = ret
+                    if not lmhash:
+                        lmhash = self.empty_lm
+                    if not nthash:
+                        nthash = self.empty_nt
+                        ## temporary fix to prevent UnicodeDecodeError backtraces 
+                        ## however this can cause truncated user names as a result
+                    name = self.get_user_name(user, samhive)
+                    if name is not None:
+                        name = name.encode('ascii', 'ignore')
+                    else:
+                        name = "(unavailable)"
```

Comment:
This should be a `BaseAbsentValue` derivative so that other plugins which may make use of the output can easily determine that the value couldn't be found, and why it couldn't be returned.